src/Neo.VM/Types/CompoundType.cs

```diff
@@ -60,12 +61,30 @@ public sealed override bool GetBoolean()
         }
 
         /// <summary>
-        /// The operation is not supported. Always throw <see cref="NotSupportedException"/>.
+        ///
+        /// This method provides a hash code for the <see cref="CompoundType"/> based on its item's span.
+        /// It is used for efficient storage and retrieval in hash-based collections.
+        ///
+        /// Use this method when you need a hash code for a <see cref="CompoundType"/>.
         /// </summary>
-        /// <exception cref="NotSupportedException">This method always throws the exception.</exception>
+        /// <returns>The hash code for the <see cref="CompoundType"/>.</returns>
         public override int GetHashCode()
         {
-            throw new NotSupportedException();
+            var h = new HashCode();
+            h.Add(Count);
+            h.Add(Type);
+            foreach (var item in SubItems)
+            {
+                try
+                {
+                    h.Add(XxHash3.HashToUInt64(item.GetSpan()));
+                }
+                catch
+                {
+                    h.Add(item.GetHashCode());
```

Comment:
It cannot handle circular references.

Reply:
GetSpan() should handle it.

Reply:
It's true that circular reference here can be problematic, we don't need an accurate hash code, fast and safe is better

Reply:
how about now? Plus i have test for that.

Reply:
> how about now? Write me a test for `circular reference` and ill fix. If isnt good enough.

Added: `TestCircularReference`

Reply:
can you check and add more test cases for circular reference. I think i have it fixed.